File: web-admin/src/services/dogAssignmentService.js
import api from './api';

export const dogAssignmentService = {
  assign: (data) => api.post('/assignments', data),
  update: (id, data) => api.put(`/assignments/${id}`, data),
  unassign: (id) => api.delete(`/assignments/${id}`),
  getById: (id) => api.get(`/assignments/${id}`),
  getByTrainer: (trainerId) => api.get(`/assignments/by-trainer/${trainerId}`),
  getByDog: (dogId) => api.get(`/assignments/by-dog/${dogId}`),
};

File: backend/src/main/java/vn/edu/fpt/doghandbook/backend/service/DogAssignmentService.java
package vn.edu.fpt.doghandbook.backend.service;

import vn.edu.fpt.doghandbook.backend.dto.request.DogAssignmentRequest;
import vn.edu.fpt.doghandbook.backend.dto.response.DogAssignmentResponse;

import java.util.List;

public interface DogAssignmentService {

    DogAssignmentResponse assign(DogAssignmentRequest request, Integer assignorId);

    DogAssignmentResponse update(Integer assignmentId, DogAssignmentRequest request);

    void unassign(Integer assignmentId);

    List<DogAssignmentResponse> getByTrainer(Integer trainerId);

    List<DogAssignmentResponse> getByDog(Integer dogId);

    DogAssignmentResponse getById(Integer assignmentId);
}
